erpnext/stock/doctype/item/item.js

```diff
@@ -170,6 +170,36 @@ frappe.ui.form.on('Item Reorder', {
 	}
 })
 
+frappe.ui.form.on('Item Customer Detail', {
+	customer_items_add: function(frm, cdt, cdn) {
+		frappe.model.set_value(cdt, cdn, 'customer_group', "");
+	},
+	customer_name: function(frm, cdt, cdn) {
+		set_customer_group(frm, cdt, cdn);
+	},
+	customer_group: function(frm, cdt, cdn) {
+		if(set_customer_group(frm, cdt, cdn)){
+			frappe.msgprint(__("Changing Customer Group for the selected Customer is not allowed."));
+		}
+	}
+});
+
+var set_customer_group = function(frm, cdt, cdn) {
+	var row = frappe.get_doc(cdt, cdn);
+	var grid_row = cur_frm.fields_dict['customer_items'].grid.grid_rows_by_docname[row.name];
+
+	if (!row.customer_name) {
+		return false;
+	}
+
+	frappe.model.with_doc("Customer", row.customer_name, function() {
+		var customer = frappe.model.get_doc("Customer", row.customer_name);
+		row.customer_group = customer.customer_group;
+		grid_row.refresh_field("customer_group");
```

Comment:
no need for `grid_row` obj here. You can use `set_value` function, it also refresh fields after setting the value.